DM-26008: Use new Config._fromPython to deserialize the pex_config files

timj opened the pull request:
Requires lsst/pex_config#59

timj committed:
Move PexConfigFormatter to obs_base

ktlim reviewed (approved):
Don't you need to add `pex_config` to the table file then, or can this only be invoked if it has already been setup somewhere else?

timj committed:
Switch formatter test to use new special formatter

Previously the test depended on PexConfigFormatter not being
loaded by any other test. This is not reliable (since it's
a real formatter) and it also required that we could always
import the formatter.  Now have a special formatter whose
entire purpose is to exist for this test.

timj commented:
I just pushed a fix to not use pexConfig in the tests.  Ideally we need to test this but that really means testing it in obs_base and moving the formatter to obs_base with all the other lsst-specific formatters. I need a new ticket for that.

timj committed:
Remove pexConfig formatter from docs

Butler doesn't know about pex_config so this can't work.
